core/dbt/exceptions.py

```diff
@@ -976,19 +976,14 @@ def warn_or_error(msg, node=None, log_fmt=None):
     if flags.WARN_ERROR:
         raise_compiler_error(msg, node)
     else:
-        if log_fmt is not None:
-            msg = log_fmt.format(msg)
-        logger.warning(msg)
+        fire_event(GeneralWarning(msg=msg, log_fmt=log_fmt))
 
 
 def warn_or_raise(exc, log_fmt=None):
     if flags.WARN_ERROR:
         raise exc
     else:
-        msg = str(exc)
-        if log_fmt is not None:
-            msg = log_fmt.format(msg)
-        logger.warning(msg)
+        fire_event(GeneralWarning(msg=str(exc), log_fmt=log_fmt))
```

Comment:
This was passing in the string form to prevent circular imports.  Had to move `pluralize` into the new `format.py` and fix all imports of `pluralize` from `utils.py` to point to `format.py`.  While all the imports are not in `/events`, they are all for purposes of logging.

Reply:
oh nice. Yeah if it's a big pain in the butt to not pass strings, it's not a hill I'll die on, but it looks like you got it work.